### ISSUE TITLE
Unsupported native payments flow?

### ISSUE THREAD

**Issue by walkerdb** [stripe/stripe-react-native#308]
**Is your feature request related to a problem? Please describe.**
First off, thank you all for all the work in putting this library together! We're very excited to migrate off of `tipsi-stripe` into an officially supported SDK.

Our company works in healthcare, which has a very asynchronous payment structure. Our current flow looks something like this:

1. a patient adds a payment method to their account from our native app. We currently manage this using `tipsi-stripe`, using either `paymentRequestWithNativePay` for Apple Pay and Android Pay cards (internally `Stripe.paymentRequestWithApplePay` or `Stripe.paymentRequestWithAndroidPay`), or with CardView + `createTokenWithCard` to create a token, then save that token as a customer source in our backend. Note that at this point we aren't actually charging them anything, we're just letting them create and persist a card using stripe.
2. the patient checks out with a medication, setting a delivery date
3. at some point between then and the delivery date we bill insurance. This could be anywhere from hours to weeks after checkout. Once we complete the insurance charge we then charge the patient's card the billed amount.

Our main requirement is that we need to be able to manage customer cards without actually creating charges during the checkout flow. 

It seems like for the  moment we can use a `SetupIntent` flow for standard cards to save the PaymentMethod to the Customer for later use (and for us to store the payment method ID to reference later). Is it possible to use the SetupIntent approach with Apple Pay (and eventually Android Pay) in a similar manner? I'm only seeing PaymentIntent approaches in the docs for those platforms so far.

**Describe the solution you'd like**
See above

**Describe alternatives you've considered**
Our payments backend is generally behind in that it's still on the Tokens and Sources flow rather than the newer PaymentIntent-based flow. We have some plans of migrating in the longer-term future but we're looking to see if there are any short-term approaches for adopting this library without also having to complete a full migration of our backend code as well.

**Comment by trex-quo** [comment]
See if this is any help to you 
https://github.com/stripe/stripe-react-native/issues/271#issuecomment-848988884

tldr; `confirmApplePayPayment` also does work with setupIntent secrets, though the documents do not explicitly say this

**Comment by thorsten-stripe** [comment]
Since v0.1.4 you can supply a `setupIntentClientSecret` to `initPaymentSheet`. We don't have dedicated docs for it yet, but it behaves exactly like https://stripe.com/docs/payments/accept-a-payment?platform=react-native&ui=payment-sheet but instead of providing `paymentIntentClientSecret` you provide `setupIntentClientSecret`.

**Comment by thorsten-stripe** [comment]
For standalone apple pay, since v0.1.4 `presentApplePay` returns a `paymentMethod` object (https://stripe.dev/stripe-react-native/api-reference/modules.html#applepayresult). You can use that to confirm the SetupIntent server-side: https://stripe.com/docs/api/setup_intents/confirm#confirm_setup_intent-payment_method

Hope this helps. Going to close this out as I think it's resolved, but please reopen if that's not the case. Thanks.
